### ISSUE TITLE
Background not showing, can't write or type

### ISSUE THREAD

**Issue by JacekJagosz** [xournalpp/xournalpp#1674]
**Affects versions :**
 - OS: MX Linux i686
 - XFCE
 - libgtk3 3.24.5
 - 1.0.16 Flatpak and 1.0.15 from MX Test repos

**Describe the bug**
The background is not showing at all, I cannot write anything, but minature show the background and I can scroll as if the page was there. I can add new pages.

**To Reproduce**
It was like that from the first lauch, persists thorugh different versions and reinstalls.
![image](https://user-images.githubusercontent.com/28653965/71007727-c7369a00-20e7-11ea-94c9-fcd69b8b59bd.png)

**Comment by LittleHuba** [comment]
> `Failed to load module "canberra-gtk-module"`

This line says everything. You are missing a gtk module on your system that is required.

**Comment by JacekJagosz** [comment]
I already had libcanberra 0.30 installed, reinstalling didn't help.

**Comment by LittleHuba** [comment]
You need to install the gtk module as well. See the instructions for Ubuntu (I have no idea if they are the same for your OS): https://askubuntu.com/questions/342202/failed-to-load-module-canberra-gtk-module-but-already-installed

**Comment by singletS** [comment]
A similar problem arises with Lubuntu 18.04, LXDE. The background sheet loads fine, but none of the tools will write on the layer.  Seems there is a problem with connecting to the jack audio server, something I'd prefer to not have to install. Tried changing audio settings under Preferences, but I don't see an option to disable audio features. I don't need audio functions; is there a way to turn them off and have xournalpp work?  Thanks.

Terminal output:

```
xournalpp 
ALSA lib pcm.c:2495:(snd_pcm_open_noupdate) Unknown PCM cards.pcm.rear
ALSA lib pcm.c:2495:(snd_pcm_open_noupdate) Unknown PCM cards.pcm.center_lfe
ALSA lib pcm.c:2495:(snd_pcm_open_noupdate) Unknown PCM cards.pcm.side
ALSA lib pcm_route.c:867:(find_matching_chmap) Found no matching channel map
ALSA lib pcm_route.c:867:(find_matching_chmap) Found no matching channel map
ALSA lib pcm_route.c:867:(find_matching_chmap) Found no matching channel map
ALSA lib pcm_route.c:867:(find_matching_chmap) Found no matching channel map
Cannot connect to server socket err = No such file or directory
Cannot connect to server request channel
jack server is not running or cannot be started
JackShmReadWritePtr::~JackShmReadWritePtr - Init not done for -1, skipping unlock
JackShmReadWritePtr::~JackShmReadWritePtr - Init not done for -1, skipping unlock
```

**Comment by JacekJagosz** [comment]
I did have those installed as well, for me both repo version and Flatpak are not working, so the possibility of missing a dependency is minimal. Could it be the fact that I'm running on 32bit?
PS MX Linux is Debian based

**Comment by LittleHuba** [comment]
> Could it be the fact that I'm running on 32bit?

Yeah that could actually be the issue. Xournal++ requires 64bit on MS Windows at least but I'm not quite sure why exactly. @andreasb242 could you shine some light on that?

I would definitely recommend you switching to 64bit anyway. Most distributions switch to a 64bit-only release system. So you will be forced to do it in one or max two years anyway. But lets first hear what @andreasb242 has to say about this. Maybe it is something completely different.

**Comment by JacekJagosz** [comment]
It definitely isn't MX Linux's problem as on newest Linux Mint 32Bit it is the same. I have been using using Xournal++ on 64Bit Solus for months, but I wanted to switch to my tiny and light netbook (Nokia Booklet 3G), so I don't have to carry my heavy laptop to classes.
I hope it is able to run on 32Bit, a lot of devices perfect for notetaking still run that

**Comment by Febbe** [comment]
At least you must compile xournal++ by yourself. The actual binaries aren't compiled for 32 bit systems.
But as far as I know xournalpp can't be compiled for 32bit systems, because there are 64bit primitives used. Maybe you need to replace them with std::ptrdiff_t/int32_t (for signed 64 types) and size_t/uint32_t (for unsigned 64 bit types).

**Comment by iamashwin99** [comment]
Hi, I have the same issue when I try to compile xournalpp for win32 bit. I have documented the process I have followed and all the warnings that I got when compiling and running the app. Can someone take a look at [my build documentation](https://github.com/iamashwin99/xournalpp/blob/master/README.md )

**Comment by Lukey3332** [comment]
Hi,
I did a quick'n dirty replace of all 'int' -> 'int64_t' and 'unsigned int' -> 'uint64_t' (So they are 64 bit even on a 32 bit cpu) and that fixed the Issue. It works very well on my 2006 fujitsu stylistic tablet. You can get the code here: https://github.com/Lukey3332/xournalpp/tree/quickndirty_int64_t
